### PR TITLE
Upgrade to OpenJDK11

### PR DESCRIPTION
**Pull request by cocreature**
I have no idea if this works but so far nobody has been able to tell me what exactly is still blocking an upgrade at this point so let’s see how it goes (I know that I also need to upgrade Windows, I’ll do that if Linux/MacOS get anywhere).
The main reason, I’m using 11 and 12 is that 12 is not yet in nixpkgs (see https://github.com/NixOS/nixpkgs/pull/58602) so doing the upgrade in two steps seems more sensible.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -35,8 +35,8 @@ build --copt -DGRPC_BAZEL_BUILD
 # Bazel defaults to JDK9. This makes sure we run Nix provided JDK, and
 # avoid dreaded "Unrecognized VM option 'CompactStrings'" error
 build --host_javabase=@java_home//:javabase
-build --java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8
-build --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8
+build --java_toolchain=@com_github_digital_asset_daml//bazel_tools:toolchain_jdk11
```
Comment (cocreature) [inline .bazelrc]: `@bazel_tools//tools/jdk:toolchain_hostjdk11` does not exist, see the comments in the definition of `toolchain_jdk11` for more information.

File: compiler/scenario-service/protos/BUILD.bazel [hunk 2]
```diff
@@ -42,6 +42,7 @@ java_library(
         "//3rdparty/jvm/io/grpc:grpc_core",
         "//3rdparty/jvm/io/grpc:grpc_protobuf",
         "//3rdparty/jvm/io/grpc:grpc_stub",
+        "//3rdparty/jvm/javax/annotation:javax_annotation_api",
```
Comment (cocreature) [inline compiler/scenario-service/protos/BUILD.bazel]: Some annotations have now moved to a separate package.

File: dependencies.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ dependencies:
   com.google.protobuf:
     protobuf-java:
       lang: java
-      version: "3.6.1"
+      version: "3.7.1"
```
Comment (cocreature) [inline dependencies.yaml]: This fixes the “illegal reflective access” warnings that we also got with older versions but it looked like we got a lot more of those with the new JDK version.

File: language-support/java/bindings/src/test/scala/com/daml/ledger/javaapi/data/TimestampSpec.scala [hunk 4]
```diff
@@ -24,14 +24,14 @@ class TimestampSpec extends FlatSpec with Matchers with GeneratorDrivenPropertyC
...
 
-  it should "be convertible to java.time.Instant" in {
-    val instant = java.time.Instant.now()
-    val timestamp = Timestamp.fromInstant(instant)
-    withClue(
-      s"input: ${instant} instant.getEpochSeconds: ${instant.getEpochSecond} instant.getNanos: ${instant.getNano} timestamp: ${timestamp} issue: ") {
-      timestamp.toInstant shouldBe instant
-    }
-  }
+  // it should "be convertible to java.time.Instant" in {
```
Comment (cocreature) [inline language-support/java/bindings/src/test/scala/com/daml/ledger/javaapi/data/TimestampSpec.scala]: Moved to #1026, I’ll rebase once that is merged.

File: language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/FromValueGenerator.scala [hunk 5]
```diff
@@ -66,7 +66,7 @@ private[inner] object FromValueGenerator extends StrictLogging {
   }
 
   private def accessors =
-    Iterator.from(0).map(i => CodeBlock.of("fields$$.get($L).getValue()", new Integer(i)))
+    Iterator.from(0).map(i => CodeBlock.of("fields$$.get($L).getValue()", Integer.valueOf(i)))
```
Comment (cocreature) [inline language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/FromValueGenerator.scala]: The constructors are now deprecated.

**Comment by cocreature** [comment]
@aherrmann-da Would be nice if you could take a brief look at the Bazel parts. CI passes now but I’ll coordinate with Spider before merging this.

**Comment by dasormeter** [comment]
hiya @aherrmann-da @moritzkiefer-da : can we go ahead with this JDK11 uptick now?  JDK 11 also makes the licensing space a bit better than JDK8 as many of the legacy gui libs that had weird licensing quirks are now gone

**Comment by cocreature** [comment]
@dasormeter No we can’t. We care still waiting for users to be ready for the update.

**Comment by stefanobaghino-da** [comment]
This is so stale it would probably be better to just close this and start over when we agree we can move on with this.

**Comment by cocreature** [comment]
But CI is green! :crying_cat_face:

**Pull request closed without merging**